### PR TITLE
chore: merge 1.4.0 into master

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Explanation:

Last week, we had to make a release to solve some **potential** security problems for users that don't customize beagle core mechanisms (network client, logger, cache) with their own implementation. Since it's impractical to have a default implementation that would satisfy completely different security concerns, we decided to remove our default implementation of these mechanisms, and let users implement their own with security concerns bounded to their specific context.

Since our master branch had features being developed, we were unable to generate a release from it. Therefore a new branch was created from the last release (1.3.0), which also incorporates hot fixes from both platforms.

Now with this PR, this code with security changes will be integrated into `master`.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
 package br.com.zup.beagle.android.cache
 
+import br.com.zup.beagle.android.logger.BeagleMessageLogs
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:19]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9940e4441ce8767eee38b5f9e6775908dfb8c464/android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt#L19"></span>The import ‘br.com.zup.beagle.android.logger.BeagleMessageLogs’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt [hunk 2]
```diff
@@ -23,11 +23,13 @@ import br.com.zup.beagle.android.context.ContextActionExecutor
 import br.com.zup.beagle.android.context.ContextData
 import br.com.zup.beagle.android.context.expressionOf
 import br.com.zup.beagle.android.context.hasExpression
+import br.com.zup.beagle.android.context.tokenizer.function.builtin.toJSONArray
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt:26]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9940e4441ce8767eee38b5f9e6775908dfb8c464/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt#L26"></span>The import ‘br.com.zup.beagle.android.context.tokenizer.function.builtin.toJSONArray’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt [hunk 3]
```diff
@@ -23,11 +23,13 @@ import br.com.zup.beagle.android.context.ContextActionExecutor
...
 import br.com.zup.beagle.android.logger.BeagleMessageLogs
 import br.com.zup.beagle.android.utils.HandleEventDeprecatedConstants.HANDLE_EVENT_ACTIONS_POINTER
 import br.com.zup.beagle.android.utils.HandleEventDeprecatedConstants.HANDLE_EVENT_DEPRECATED_MESSAGE
 import br.com.zup.beagle.android.utils.HandleEventDeprecatedConstants.HANDLE_EVENT_POINTER
 import br.com.zup.beagle.android.widget.RootView
+import java.lang.reflect.Type
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt:32]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9940e4441ce8767eee38b5f9e6775908dfb8c464/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ActionExtensions.kt#L32"></span>The import ‘java.lang.reflect.Type’ is unused.</td>
    </tr>
  </tbody>
</table>

File: android/sample/src/main/java/br/com/zup/beagle/sample/config/HttpClientDefault.kt [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+package br.com.zup.beagle.sample.config
+
+import br.com.zup.beagle.android.annotation.BeagleComponent
+import br.com.zup.beagle.android.exception.BeagleApiException
+import br.com.zup.beagle.android.networking.HttpClient
+import br.com.zup.beagle.android.networking.HttpMethod
+import br.com.zup.beagle.android.networking.RequestCall
+import br.com.zup.beagle.android.networking.RequestData
+import br.com.zup.beagle.android.networking.ResponseData
+import br.com.zup.beagle.sample.config.CoroutineDispatchers
```
Comment (botbeaglezup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/config/HttpClientDefault.kt:26]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9940e4441ce8767eee38b5f9e6775908dfb8c464/android/sample/src/main/java/br/com/zup/beagle/sample/config/HttpClientDefault.kt#L26"></span>The import ‘br.com.zup.beagle.sample.config.CoroutineDispatchers’ is unused.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/3044471fa465ecc438a26caa234d7273ee09eccf/android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/PolymorphicJsonAdapterFactory.java#L308">android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/PolymorphicJsonAdapterFactory.java#L308</a> - Implicitly using the default locale is a common source of bugs: Use <code>toLowerCase(Locale)</code> instead. For strings meant to be internal use <code>Locale.ROOT</code>, otherwise <code>Locale.getDefault()</code>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by viniciusguardieirozup — APPROVED** [review]

**Review by hernandazevedozup — APPROVED** [review]

**Review by theffc — APPROVED** [review]

**Pull request merged**